src/core/Cline.ts

```diff
@@ -3120,6 +3141,42 @@ export class Cline {
 
 		telemetryService.captureConversationTurnEvent(this.taskId, this.apiProvider, this.api.getModel().id, "user")
 
+		// Capture message data for telemetry,
+		// ONLY if user is opted in, in advanced settings
+		// This is done after a timeout to ensure the message is added to the conversation history
+		setTimeout(() => {
+			if (this.apiProvider && this.api.getModel().id) {
+				const metadata = {
+					apiProvider: this.apiProvider,
+					model: this.api.getModel().id,
+					tokensIn: 0,
+					tokensOut: 0,
+				}
+
+				// Get the last message from apiConversationHistory
+				const lastMessage = this.apiConversationHistory[this.apiConversationHistory.length - 1]
+
+				// Get the corresponding timestamp from clineMessages
+				// The last message in clineMessages should be the one we just added
+
+				const lastClineMessage = this.clineMessages[this.clineMessages.length - 1]
+				const ts = lastClineMessage.ts
+
+				// Add the timestamp to the message object for telemetry
+				const messageWithTs = {
+					...lastMessage,
+					ts,
+				}
+
+				// Send individual message to telemetry
+				conversationTelemetryService.captureMessage(this.taskId, messageWithTs, metadata)
+
+				// Send entire conversation history to cleanup endpoint
+				// This ensures deleted messages are properly handled in telemetry
+				conversationTelemetryService.cleanupTask(this.taskId, this.clineMessages)
+			}
+		}, 5)
```

Comment:
We should consider not providing a timeout time here.  setTimeout when you don't pass a time for the timeout just appends the predicate to the bottom of the stack.  We may be fine with this just being appended to the bottom of the stack.